contracts/Aavegotchi/ForgeDiamond/facets/ForgeVRFFacet.sol

```diff
@@ -0,0 +1,252 @@
+        VrfRequestInfo storage info = s.vrfRequestIdToVrfRequestInfo[_requestId];
+
+        require(s.userVrfPending[info.user], "ForgeVRFFacet: VRF is not pending for user");
+        require(info.status == VrfStatus.PENDING, "ForgeVRFFacet: VRF request is not pending");
+
+        info.randomNumber = _randomNumber;
+        info.status = VrfStatus.READY_TO_CLAIM;
+
+        emit VrfResponse(info.user, _randomNumber, _requestId, block.number);
+    }
+
+    /**
+     * @notice fulfillRandomness handles the VRF response. Your contract must
+     * @notice implement it.
+     *
+     * @dev The VRFCoordinator expects a calling contract to have a method with
+     * @dev this signature, and will trigger it once it has verified the proof
+     * @dev associated with the randomness (It is triggered via a call to
+     * @dev rawFulfillRandomness, below.)
+     *
+     * @param _requestId The Id initially returned by requestRandomness
+     * @param _randomNumber the VRF output
+//     */
+    function rawFulfillRandomness(bytes32 _requestId, uint256 _randomNumber) external {
+        require(LibMeta.msgSender() == s.vrfCoordinator, "Only VRFCoordinator can fulfill");
+
+        VrfRequestInfo storage info = s.vrfRequestIdToVrfRequestInfo[_requestId];
+
+        require(s.userVrfPending[info.user], "ForgeVRFFacet: VRF is not pending for user");
+        require(info.status == VrfStatus.PENDING, "ForgeVRFFacet: VRF request is not pending");
+
+        info.randomNumber = _randomNumber;
+        info.status = VrfStatus.READY_TO_CLAIM;
+
+        emit VrfResponse(info.user, _randomNumber, _requestId, block.number);
+    }
+
+    function getRequestInfo(address user) external view returns (VrfRequestInfo memory) {
+        //        require(s.vrfUserToRequestIds[user].length > 0, "ForgeVRFFacet: No VRF requests");
+        bytes32 requestId = s.vrfUserToRequestIds[user][s.vrfUserToRequestIds[user].length - 1];
+        VrfRequestInfo storage info = s.vrfRequestIdToVrfRequestInfo[requestId];
+
+        return info;
+    }
+
+    function getRequestInfoByRequestId(bytes32 requestId) external view returns (VrfRequestInfo memory) {
+        return s.vrfRequestIdToVrfRequestInfo[requestId];
+    }
+
+    function claimWinnings() external whenNotPaused {
+        address sender = LibMeta.msgSender();
+
+        require(s.vrfUserToRequestIds[sender].length > 0, "ForgeVRFFacet: No VRF requests");
+        bytes32 requestId = s.vrfUserToRequestIds[sender][s.vrfUserToRequestIds[sender].length - 1];
+
+        VrfRequestInfo storage info = s.vrfRequestIdToVrfRequestInfo[requestId];
+
+        require(info.status == VrfStatus.READY_TO_CLAIM, "ForgeVRFFacet: not ready to claim");
+        require(info.randomNumber != 0, "ForgeVRFFacet: invalid random number");
+
+        uint256 modNum = 10000;
+        uint256 divNum = 1;
+
+        uint256 prizesLeft = numTotalPrizesLeft();
+        uint256 numWins;
```

Comment:
Should `numWins` get reset when you set geode prizes? If it keeps incrementing, there could eventually be a situation where it gets higher than the `prizesLeft`, even though you just reset with new prizes. @infiniRekt